docs/examples.circuit_relay.rst

```diff
@@ -0,0 +1,317 @@
+Circuit Relay v2 Example
+========================
+
+This example demonstrates how to use Circuit Relay v2 in py-libp2p. It includes three components:
+
+1. A relay node that provides relay services
+2. A destination node that accepts relayed connections
+3. A source node that connects to the destination through the relay
+
+Prerequisites
+-------------
+
+First, ensure you have py-libp2p installed:
+
+.. code-block:: console
+
+    $ python -m pip install libp2p
+    Collecting libp2p
+    ...
+    Successfully installed libp2p-x.x.x
+
+Relay Node
+----------
+
+Create a file named ``relay_node.py`` with the following content:
+
+.. code-block:: python
+
+    import trio
+    import logging
+    import multiaddr
+
+    from libp2p import new_host
+    from libp2p.relay.circuit_v2.protocol import CircuitV2Protocol
+    from libp2p.relay.circuit_v2.transport import CircuitV2Transport
+    from libp2p.relay.circuit_v2.config import RelayConfig
+
+    # Configure logging
+    logging.basicConfig(level=logging.DEBUG)
+
+    async def run_relay():
+        # Create a libp2p host
+        host = await new_host(
```

Comment:
I'm getting `TypeError: object BasicHost can't be used in 'await' expression` here.